### PR TITLE
TST: add workflow_dispatch for pysat RC

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses pysat/pysat#1042

Adds a dispatch workflow to test this package with the latest pysat RC.

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Inspecting the logs at https://github.com/pysat/pysatMadrigal/actions/runs/3440277429

Ensure that pysat is installed only from the test pypi server.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: .github/workflows/pysat_rc.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+name: Test with latest pysat RC
+
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (aburrell) [inline .github/workflows/pysat_rc.yml:13]: Is there a TODO for windows support?

Comment (aburrell) [reply .github/workflows/pysat_rc.yml:13]: So, I added windows CI tests to the other active PR and they broke.  Issue #76 has been created to capture the problem.

Comment (jklenzing) [reply .github/workflows/pysat_rc.yml:13]: Do I need to update anything here for the windows issues?

Comment (aburrell) [reply .github/workflows/pysat_rc.yml:13]: Just add the TODO for #76.

Comment (jklenzing) [inline .github/workflows/pysat_rc.yml:13]:
```suggestion
# TODO(#76): add windows tests when bugs are sorted
        os: [ubuntu-latest, macos-latest]
```

**Comment by jklenzing** [comment]
Woohoo! All tests finally passing!

**Review by aburrell — APPROVED** [review]

**Pull request merged**